### PR TITLE
snippets(mc): #3078 Add defaultBrowser, selectedSearchEngine to snippets

### PR DESCRIPTION
**Pull request by k88hudson**

### CODE REVIEW

File: system-addon/lib/SnippetsFeed.jsm [hunk 1]
```diff
@@ -27,12 +33,22 @@ this.SnippetsFeed = class SnippetsFeed {
...
       .replace("%STARTPAGE_VERSION%", STARTPAGE_VERSION);
     return Services.urlFormatter.formatURL(updateURL);
   }
+
+  isDefaultBrowser() {
+    try {
+      return ShellService.isDefaultBrowser();
+    } catch (e) {}
+    // istanbul ignore next
+    return null;
```
Comment (sarracini) [inline system-addon/lib/SnippetsFeed.jsm]: nit: indentation is off i think. also, do we want to return false for consistency or is it necessary to return null for some other part of the code to know it went wrong?

File: docs/v2-system-addon/snippets.md [hunk 2]
```diff
@@ -26,28 +26,50 @@ async function setFoo() {
...
+{
+  engines: ["searchEngine-google", "searchEngine-yahoo", "searchEngine-bing"],
+  searchEngineIdentifier : "google"
+}
+```
 
-`appData.telemetryEnabled`: Is telemetry enabled for the user?
+`appData.defaultBrowser`: (bool) Is Firefox the user's default browser?
+If we could not determine the default browser, this value is `null`. Note that
+currently this value is only checked once when the browser is initialized.
```
Comment (sarracini) [inline docs/v2-system-addon/snippets.md]: how does this work? is ```_refresh``` only called on browser init?

File: system-addon/lib/SnippetsFeed.jsm (hunk 1 above)
Comment (k88hudson) [inline system-addon/lib/SnippetsFeed.jsm]: There needs to be a distinction between `true`, `false`, and we couldn't actually determine the value (this is also copying what the existing API is)

File: docs/v2-system-addon/snippets.md (hunk 2 above)
Comment (k88hudson) [inline docs/v2-system-addon/snippets.md]: Yes that's right

File: system-addon/lib/SnippetsFeed.jsm (hunk 1 above)
Comment (k88hudson) [inline system-addon/lib/SnippetsFeed.jsm]: Where's the indentation issue? the return null is outside the catch block

Comment (sarracini) [inline system-addon/lib/SnippetsFeed.jsm]: whoooops my bad thought it was in the catch

**Comment by as-pine-proxy** [comment]
[Treeherder: 4a12df4c93a2ff9f1127ba861178d040995de02d](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=4a12df4c93a2ff9f1127ba861178d040995de02d)

**Pull request merged**